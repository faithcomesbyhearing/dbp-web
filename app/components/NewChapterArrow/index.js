/**
 *
 * NewChapterArrow
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { FormattedMessage } from 'react-intl';
import SvgWrapper from '../SvgWrapper';

function NewChapterArrow({
  clickHandler,
  getNewUrl,
  disabled,
  svgid,
  svgClasses,
  containerClasses,
  urlProps,
  disabledContainerClasses,
  title,
  id,
  textProps,
}) {
  if (disabled) {
    return (
      <div className={disabledContainerClasses || 'arrow-wrapper disabled'} />
    );
  }

  return (
    <Link
      legacyBehavior
      as={getNewUrl({
        ...urlProps,
        isHref: false,
      })}
      href={getNewUrl({
        ...urlProps,
        isHref: true,
      })}
    >
      <div
        onClick={clickHandler}
        className={containerClasses || 'arrow-wrapper'}
        title={title || ''}
        id={id || ''}
      >
        <SvgWrapper className={svgClasses || 'arrow-svg'} svgid={svgid} />
        {!!textProps && <FormattedMessage {...textProps} />}
      </div>
    </Link>
  );
}

NewChapterArrow.propTypes = {
  getNewUrl: PropTypes.func,
  disabled: PropTypes.bool,
  id: PropTypes.string,
  title: PropTypes.string,
  svgid: PropTypes.string,
  svgClasses: PropTypes.string,
  containerClasses: PropTypes.string,
  disabledContainerClasses: PropTypes.string,
  urlProps: PropTypes.object,
  textProps: PropTypes.object,
  clickHandler: PropTypes.func,
};

export default NewChapterArrow;
