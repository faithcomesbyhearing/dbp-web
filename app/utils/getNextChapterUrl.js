import url from './hrefLinkOrAsLink';

export default ({
	books,
	chapter,
	bookId,
	textId,
	verseNumber,
	text: chapterText,
	isHref,
	audioType,
}) => {
	if (!books || !books.length) {
		return url({ audioType, textId, bookId, chapter, isHref });
	}
	if (verseNumber && chapterText.length) {
		const nextVerse = parseInt(verseNumber, 10) + 1 || 1;
		const lastVerse = chapterText.length;

		// Handles the verses
		if (nextVerse <= lastVerse && nextVerse > 0) {
			// The next verse is within a valid range
			return url({ audioType, textId, bookId, chapter, nextVerse, isHref });
		} else if (nextVerse < 0) {
			// The next verse is below 0 and thus invalid
			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: '1',
				isHref,
			});
		} else if (nextVerse > lastVerse) {
			// Next verse is above the last verse in the chapter and thus is invalid
			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: lastVerse,
				isHref,
			});
		}
	} else if (verseNumber) {
		const nextVerse = parseInt(verseNumber, 10) + 1 || 1;

		if (nextVerse && nextVerse > 0) {
			// The next verse is within a valid range
			return url({ audioType, textId, bookId, chapter, nextVerse, isHref });
		} else if (nextVerse < 0) {
			// The next verse is below 0 and thus invalid

			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: '1',
				isHref,
			});
			// Need to find a way to do this for formatted text
			// Next verse is above the last verse in the chapter and thus is invalid
		}
	}

	// Handles the chapters
	const activeBook = books.find(
		(book) => book.book_id.toLowerCase() === bookId.toLowerCase(),
	);
	// This relies on the books list already being sorted in ascending book_order
	const nextBook = books.find(
		(book) =>
			activeBook &&
			(book.book_order === activeBook.book_order + 1 ||
				book.book_order > activeBook.book_order),
	);
	const maxChapter = activeBook?.chapters[activeBook.chapters.length - 1];
	// If the next book in line doesn't exist and we are already at the last chapter just return
	if (!nextBook && chapter === maxChapter) {
		return url({ audioType, textId, bookId, chapter });
	}

	if (chapter === maxChapter) {
		// Need to get the first chapter of the next book
		return url({
			audioType,
			textId,
			bookId: nextBook.book_id,
			chapter: nextBook.chapters[0],
			isHref,
		});
	}

	const chapterIndex = activeBook?.chapters.findIndex(
		(c) => c === chapter || c > chapter,
	);
	const nextChapterNumber =
		activeBook?.chapters[chapterIndex] === chapter
			? activeBook?.chapters[chapterIndex + 1]
			: activeBook?.chapters[chapterIndex];

	if (!nextChapterNumber) {
		return url({ audioType, textId, bookId, chapter, isHref });
	}

	return url({
		audioType,
		textId,
		bookId,
		chapter: nextChapterNumber,
		isHref,
	});
};
