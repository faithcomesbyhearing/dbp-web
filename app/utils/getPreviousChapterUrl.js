import url from './hrefLinkOrAsLink';

export default ({
	books,
	chapter,
	bookId,
	textId,
	verseNumber,
	text: chapterText,
	isHref,
	audioType,
}) => {
	if (!books?.length) {
		return url({ audioType, textId, bookId, chapter, isHref });
	}
	if (verseNumber && chapterText.length) {
		const prevVerse = parseInt(verseNumber, 10) - 1 || 1;
		const lastVerse = chapterText.length;
		// Is it past the max verses for the chapter?
		// if not increment it by 1
		if (prevVerse <= lastVerse && prevVerse > 0) {
			// Verse was valid

			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: prevVerse,
				isHref,
			});
		} else if (prevVerse < 0) {
			// Verse was below valid range

			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: '1',
				isHref,
			});
		} else if (prevVerse > lastVerse) {
			// Verse was above valid range

			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: lastVerse,
				isHref,
			});
		}
	} else if (verseNumber) {
		const nextVerse = parseInt(verseNumber, 10) - 1 || 1;

		if (nextVerse && nextVerse > 0) {
			// The next verse is within a valid range

			return url({ audioType, textId, bookId, chapter, nextVerse, isHref });
		} else if (nextVerse < 0) {
			// The next verse is below 0 and thus invalid

			return url({
				audioType,
				textId,
				bookId,
				chapter,
				nextVerse: '1',
				isHref,
			});
		}
	}

	if (
		books[0] &&
		bookId.toLowerCase() === books[0].book_id.toLowerCase() &&
		chapter - 1 === 0
	) {
		return url({ audioType, textId, bookId, chapter, isHref });
	}

	let activeBookIndex;
	let previousBookIndex;

	books.forEach((book, index) => {
		if (book.book_id.toLowerCase() === bookId.toLowerCase()) {
			activeBookIndex = index;

			if (index - 1 >= 0) {
				previousBookIndex = index - 1;
			} else {
				previousBookIndex = index;
			}
		}
	});

	const previousBook = books[previousBookIndex] ?? {
		chapters: [],
		book_id: '',
	};
	const activeBook = books[activeBookIndex] ?? { chapters: [], book_id: '' };
	if (chapter - 1 === 0) {
		const chapters = Array.isArray(previousBook.chapters)
			? previousBook.chapters
			: [];

		const lastChapter =
			(chapters.length && chapters[chapters.length - 1]) || chapter;
		// Goes to the last chapter in the previous book
		return url({
			audioType,
			textId,
			bookId: previousBook['book_id'],
			// chapter: previousBook?.['chapters']?.[-1],
			chapter: lastChapter,
			isHref,
		});
	}
	// If the chapter number is greater than the active chapter then we have gone too far and need to get the previous chapter
	const chapterIndex = activeBook['chapters'].findIndex(
		(c) => c === chapter || c > chapter,
	);

	if (!chapterIndex) {
		return url({ audioType, textId, bookId, chapter, isHref });
	}

	return url({ audioType, textId, bookId, chapter: chapterIndex, isHref });
};
