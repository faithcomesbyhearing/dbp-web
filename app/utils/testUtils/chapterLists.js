export const textChapters = {
	ENGKJV: {
		filesetType: 'text_plain',
		books: [
			{
				book_id: 'GEN',
				book_id_usfx: 'GN',
				book_id_osis: 'Gen',
				name: 'Genesis',
				testament: 'OT',
				testament_order: 1,
				book_order: 1,
				book_group: 'The Law',
				chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
			},
			{
				book_id: 'EXO',
				book_id_usfx: 'EX',
				book_id_osis: 'Exod',
				name: 'Exodus',
				testament: 'OT',
				testament_order: 2,
				book_order: 2,
				book_group: 'The Law',
				chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
			},
			{
				book_id: 'MAT',
				book_id_usfx: 'MT',
				book_id_osis: 'Matt',
				name: 'Matthew',
				testament: 'NT',
				testament_order: 1,
				book_order: 41,
				book_group: 'Gospels',
				chapters: [11],
			},
			{
				book_id: 'MRK',
				book_id_usfx: 'MK',
				book_id_osis: 'Mark',
				name: 'Mark',
				testament: 'NT',
				testament_order: 2,
				book_order: 42,
				book_group: 'Gospels',
				chapters: [15],
			},
			{
				book_id: 'LUK',
				book_id_usfx: 'LK',
				book_id_osis: 'Luke',
				name: 'Luke',
				testament: 'NT',
				testament_order: 3,
				book_order: 43,
				book_group: 'Gospels',
				chapters: [7, 8, 9, 10],
			},
			{
				book_id: 'JHN',
				book_id_usfx: 'JN',
				book_id_osis: 'John',
				name: 'John',
				testament: 'NT',
				testament_order: 4,
				book_order: 44,
				book_group: 'Gospels',
				chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
			},
		],
	},
};
export const audioChapters = {
	ENGKJVN2DA: {
		filesetType: 'audio_drama',
		books: [
			{
				book_id: 'MAT',
				book_id_usfx: 'MT',
				book_id_osis: 'Matt',
				name: 'Matthew',
				testament: 'NT',
				testament_order: 1,
				book_order: 41,
				book_group: 'Gospels',
				chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
			},
			{
				book_id: 'MRK',
				book_id_usfx: 'MK',
				book_id_osis: 'Mark',
				name: 'Mark',
				testament: 'NT',
				testament_order: 2,
				book_order: 42,
				book_group: 'Gospels',
				chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
			},
			{
				book_id: 'LUK',
				book_id_usfx: 'LK',
				book_id_osis: 'Luke',
				name: 'Luke',
				testament: 'NT',
				testament_order: 3,
				book_order: 43,
				book_group: 'Gospels',
				chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
			},
		],
	},
};
