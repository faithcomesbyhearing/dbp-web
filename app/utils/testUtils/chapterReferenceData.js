export const filesets = [
	{ id: 'ENGESVN2DA', type: 'audio_drama', size: 'NT' },
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESVO2DA', type: 'audio_drama', size: 'OT' },
	{ id: 'ENGESVO1DA', type: 'audio', size: 'OT' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
	{ id: 'ENGESVN1DA', type: 'audio', size: 'NT' },
	{ id: 'ENGESVP2DV', type: 'video_stream', size: 'NTP' },
];
export const filesets2 = [
	{ id: 'ENGESVN2DA', type: 'audio_drama', size: 'NT' },
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESVO2DA', type: 'audio_drama', size: 'OT' },
	{ id: 'ENGESVO1DA', type: 'audio', size: 'OT' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
	{ id: 'ENGESVN1DA', type: 'audio', size: 'NT' },
];
export const filesets3 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESVO2DA', type: 'audio_drama', size: 'OT' },
	{ id: 'ENGESVO1DA', type: 'audio', size: 'OT' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
];
export const filesets4 = [
	{ id: 'ENGESVN2DA', type: 'audio_drama', size: 'NT' },
	{ id: 'ENGESVN1DA', type: 'audio', size: 'NT' },
];
export const filesets5 = [
	{ id: 'ENGESVO2DA', type: 'audio_drama', size: 'OT' },
	{ id: 'ENGESVO1DA', type: 'audio', size: 'OT' },
];
export const filesets6 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
];
export const filesets7 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
];
export const filesets8 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'OT' },
	{ id: 'ENGESV', type: 'text_format', size: 'OT' },
];
export const bookMetaResponse = [
	{
		ENGESVP2DV: {
			filesetType: 'video_stream',
			books: [
				{
					book_id: 'MRK',
					book_id_usfx: 'MR',
					book_id_osis: 'Mrk',
					name: 'Mark',
					testament: 'NT',
					testament_order: 2,
					book_order: 42,
					book_group: 'Gospels',
					chapters: [6, 7],
				},
			],
		},
	},
	{
		ENGESVN2DA: {
			filesetType: 'audio_drama',
			books: [
				{
					book_id: 'MAT',
					book_id_usfx: 'GN',
					book_id_osis: 'Mat',
					name: 'Matthew',
					testament: 'NT',
					testament_order: 1,
					book_order: 41,
					book_group: 'Gospels',
					chapters: [5, 6, 7],
				},
			],
		},
	},
	{
		ENGESV: {
			filesetType: 'text_plain',
			books: [
				{
					book_id: 'GEN',
					book_id_usfx: 'GN',
					book_id_osis: 'Gen',
					name: 'Genesis',
					testament: 'OT',
					testament_order: 1,
					book_order: 1,
					book_group: 'The Law',
					chapters: [1, 2, 3, 4, 5, 6, 7],
				},
				{
					book_id: 'MAT',
					book_id_usfx: 'GN',
					book_id_osis: 'Mat',
					name: 'Matthew',
					testament: 'NT',
					testament_order: 1,
					book_order: 41,
					book_group: 'Gospels',
					chapters: [1, 2, 3, 4, 5, 6, 7],
				},
				{
					book_id: 'PSA',
					book_id_usfx: 'PS',
					book_id_osis: 'Psa',
					name: 'Psalms',
					testament: 'OT',
					testament_order: 1,
					book_order: 29,
					book_group: 'The Law',
					chapters: [70],
				},
			],
		},
	},
	{
		ENGESV: {
			filesetType: 'text_format',
			books: [
				{
					book_id: 'PSA',
					book_id_usfx: 'PS',
					book_id_osis: 'Psa',
					name: 'Psalms',
					testament: 'OT',
					testament_order: 1,
					book_order: 29,
					book_group: 'The Law',
					chapters: [70],
				},
			],
		},
	},
	{
		ENGESVO2DA: {
			filesetType: 'audio_drama',
			books: [
				{
					book_id: 'PSA',
					book_id_usfx: 'PS',
					book_id_osis: 'Psa',
					name: 'Psalms',
					testament: 'OT',
					testament_order: 1,
					book_order: 29,
					book_group: 'The Law',
					chapters: [6, 70],
				},
			],
		},
	},
	{
		ENGESVO1DA: {
			filesetType: 'audio',
			books: [
				{
					book_id: 'GEN',
					book_id_usfx: 'GN',
					book_id_osis: 'Gen',
					name: 'Genesis',
					testament: 'OT',
					testament_order: 1,
					book_order: 1,
					book_group: 'The Law',
					chapters: [1, 2, 3, 4, 5, 6, 7],
				},
			],
		},
	},
	{
		ENGESVN1DA: {
			filesetType: 'audio',
			books: [
				{
					book_id: 'MAT',
					book_id_usfx: 'GN',
					book_id_osis: 'Mat',
					name: 'Matthew',
					testament: 'NT',
					testament_order: 1,
					book_order: 41,
					book_group: 'Gospels',
					chapters: [5, 6, 7],
				},
			],
		},
	},
];
export const bookMetaData = [
	{
		book_id: 'GEN',
		book_id_usfx: 'GN',
		book_id_osis: 'Gen',
		name: 'Genesis',
		testament: 'OT',
		testament_order: 1,
		book_order: 1,
		book_group: 'The Law',
		chapters: [1, 2, 3, 4, 5, 6, 7, 48, 49, 50],
	},
];
