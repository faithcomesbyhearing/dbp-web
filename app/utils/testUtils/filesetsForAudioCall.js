export const engesv = [
	{
		id: 'ENGESVN2DA',
		type: 'audio_drama',
		size: 'NT',
	},
	{
		id: 'ENGESV',
		type: 'text_plain',
		size: 'C',
	},
	{
		id: 'ENGESVC2DA',
		type: 'audio_drama',
		size: 'OT',
	},
	{
		id: 'ENGESVO2DA',
		type: 'audio_drama',
		size: 'OT',
	},
	{
		id: 'ENGESVO1DA',
		type: 'audio',
		size: 'OT',
	},
	{
		id: 'ENGESV',
		type: 'text_format',
		size: 'C',
	},
	{
		id: 'ENGESVN1DA',
		type: 'audio',
		size: 'NT',
	},
];
export const ntpotp = [
	{
		id: 'OREWBTP1DA',
		type: 'audio',
		size: 'OTP',
	},
	{
		id: 'OREWBTS1DA',
		type: 'audio',
		size: 'OTP',
	},
	{
		id: 'OREWBTP2DA',
		type: 'audio_drama',
		size: 'NTP',
	},
];
export const seawbt = [
	{
		id: 'SEAWYIP2DA',
		type: 'audio_drama',
		size: 'NTPOTP',
	},
	{
		id: 'SEAWYIP2DA',
		type: 'app',
		size: 'NTP',
	},
	{
		id: 'SEAWYIP1DA',
		type: 'audio',
		size: 'NTPOTP',
	},
	{
		id: 'SEAWYIP2DA16',
		type: 'audio_drama',
		size: 'NTPOTP',
	},
];
export const filesets3 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESVO2DA', type: 'audio_drama', size: 'OT' },
	{ id: 'ENGESVO1DA', type: 'audio', size: 'OT' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
];
export const filesets4 = [
	{ id: 'ENGESVN2DA', type: 'audio_drama', size: 'NT' },
	{ id: 'ENGESVN1DA', type: 'audio', size: 'NT' },
];
export const filesets5 = [
	{ id: 'ENGESVO2DA', type: 'audio_drama', size: 'OT' },
	{ id: 'ENGESVO1DA', type: 'audio', size: 'OT' },
];
export const filesets6 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
];
export const filesets7 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'C' },
	{ id: 'ENGESV', type: 'text_format', size: 'C' },
];
export const filesets8 = [
	{ id: 'ENGESV', type: 'text_plain', size: 'OT' },
	{ id: 'ENGESV', type: 'text_format', size: 'OT' },
];
