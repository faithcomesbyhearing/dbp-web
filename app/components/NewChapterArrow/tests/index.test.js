import React from 'react';
import { render } from '@testing-library/react';
import NewChapterArrow from '..';
import getPreviousChapterUrl from '../../../utils/getPreviousChapterUrl';
import getNextChapterUrl from '../../../utils/getNextChapterUrl';
import { text } from '../../../utils/testUtils/plaintextData';
import { bookData } from '../../../utils/testUtils/booksData';

const books = [...bookData.OT, ...bookData.NT];

const prevProps = {
	disabled: false,
	clickHandler: jest.fn(),
	svgid: 'arrow_left',
	svgClasses: 'prev-arrow-svg',
	disabledContainerClasses: 'arrow-wrapper prev disabled',
	containerClasses: 'arrow-wrapper prev',
	getNewUrl: getPreviousChapterUrl,
	urlProps: {
		books,
		text,
		chapter: 2,
		bookId: 'mat',
		textId: 'engesv',
		verseNumber: 0,
		audioType: 'audio_drama',
	},
};
const nextProps = {
	disabled: false,
	clickHandler: jest.fn(),
	svgid: 'arrow_left',
	svgClasses: 'next-arrow-svg',
	disabledContainerClasses: 'arrow-wrapper next disabled',
	containerClasses: 'arrow-wrapper next',
	getNewUrl: getNextChapterUrl,
	urlProps: {
		books,
		text,
		chapter: 2,
		bookId: 'mat',
		textId: 'engesv',
		verseNumber: 0,
		audioType: 'audio_drama',
	},
};

describe('NewChapterArrow', () => {
	it('should match snapshot for previous arrow base props', () => {
		const { asFragment } = render(<NewChapterArrow {...prevProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('should match snapshot for next arrow base props', () => {
		const { asFragment } = render(<NewChapterArrow {...nextProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('should match snapshot for previous arrow disabled props', () => {
		const { asFragment } = render(<NewChapterArrow {...prevProps} disabled />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('should match snapshot for next arrow disabled props', () => {
		const { asFragment } = render(<NewChapterArrow {...nextProps} disabled />);
		expect(asFragment()).toMatchSnapshot();
	});
});
