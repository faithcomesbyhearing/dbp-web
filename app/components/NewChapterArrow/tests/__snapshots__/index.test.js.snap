// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewChapterArrow should match snapshot for next arrow base props 1`] = `
<DocumentFragment>
  <a
    class="arrow-wrapper next"
    href="/bible/engesv/mat/3?audio_type=audio_drama"
    title=""
  >
    <svg
      class="next-arrow-svg"
    >
      <use
        xlink:href="/svglist.svg#arrow_left"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`NewChapterArrow should match snapshot for next arrow disabled props 1`] = `
<DocumentFragment>
  <div
    class="arrow-wrapper next disabled"
  />
</DocumentFragment>
`;

exports[`NewChapterArrow should match snapshot for previous arrow base props 1`] = `
<DocumentFragment>
  <a
    class="arrow-wrapper prev"
    href="/bible/engesv/mat/1?audio_type=audio_drama"
    title=""
  >
    <svg
      class="prev-arrow-svg"
    >
      <use
        xlink:href="/svglist.svg#arrow_left"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`NewChapterArrow should match snapshot for previous arrow disabled props 1`] = `
<DocumentFragment>
  <div
    class="arrow-wrapper prev disabled"
  />
</DocumentFragment>
`;
